### PR TITLE
Smarter nodes 2 (excluding file differences; do not merge)

### PR DESCRIPTION
**Pull request by enzuru**

### CODE REVIEW

**Comment by hjr3** [comment]
@enzuru awesome that you are writing some scala now!

File: src/main/scala/solr/supervisor/Watch.scala [hunk 1]
```diff
@@ -65,104 +97,132 @@ object ZK {
...
     logger.info(event.toString)
-    Thread.sleep(5000l) // been getting error 500s, maybe going too fast?
+    Thread.sleep(5000l)
+
     deadNode ! GetChildren(event.getPath)
   }
 }
 
-class CollectionsActor extends Actor with LazyLogging {
+class CollectionActor extends Actor with LazyLogging {
```
Comment (jpswain) [inline src/main/scala/solr/supervisor/Watch.scala]: @enzuru i think this needs to be Collections (plural) b/c we are working with the "Collections API" in SolrCloud.

File: src/main/scala/solr/supervisor/Watch.scala [hunk 2]
```diff
@@ -65,104 +97,132 @@ object ZK {
...
       self ! Reap(liveNodes, state.groupBy(_.node))
-      self ! Sow(liveNodes, collectionIndexed, ZK.collections.toList)
+      self ! Sow(liveNodes, collectionIndexed, ZooKeeper.collectionsState.toList)
   }
 
-  // this should be safe to run multiple times over and over
   def reap(liveNodes: Set[String], nodeMap: Map[String, Seq[Record]]) = {
-    val deadNodes = nodeMap.filterKeys(k => !liveNodes.contains(k)).values.flatten
-    deadNodes.foreach({ case Record(_, core, shard, coll, _) => Capi(coll).delete(shard, core) })
+    // Safe to run multiple times
```
Comment (jpswain) [inline src/main/scala/solr/supervisor/Watch.scala]: @enzuru i think these comments apply to the whole method, so they should be outside, the line above "def reap", in this case

File: src/main/scala/solr/supervisor/Watch.scala [hunk 3]
```diff
@@ -16,48 +16,80 @@ import scala.concurrent.duration._
...
+
+  lazy val SystemHostZooKeeper = System.getProperty("solr.zk")
+  lazy val SystemHostSolr = System.getProperty("solr.host")
+
+  lazy val hostZooKeeper = Option(SystemHostZooKeeper).getOrElse(DefaultHostZooKeeper)
+  lazy val hostSolr = Option(SystemHostSolr).getOrElse(DefaultHostSolr)
 }
 
-case class CollectionsApi(solrHost: String, collection: String) {
+case class CollectionApi(hostSolr: String, collection: String) {
```
Comment (jpswain) [inline src/main/scala/solr/supervisor/Watch.scala]: @enzuru i think this needs to be Collections (plural) b/c we are working with the "Collections API" in SolrCloud.
 Add a line note

**Pull request closed without merging**